diff --git a/libdiscord.c b/libdiscord.c
index 0c448d7..99b77de 100644
--- a/libdiscord.c
+++ b/libdiscord.c
@@ -3513,7 +3513,7 @@ discord_find_chat_in_group(PurpleAccount *account, const char *id, PurpleGroup *
 
 
 static void
-discord_add_channel_to_blist(DiscordAccount *da, DiscordChannel *channel, PurpleGroup *group)
+discord_add_channel_to_blist(DiscordAccount *da, DiscordChannel *channel, DiscordGuild *guild, PurpleGroup *group)
 {
 	GHashTable *components = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
 	gchar *id = from_int(channel->id);
@@ -3525,9 +3525,17 @@ discord_add_channel_to_blist(DiscordAccount *da, DiscordChannel *channel, Purple
 		group = discord_get_or_create_default_group();
 	}
 
+	DiscordChannel *cat = NULL;
+	if (guild)
+		cat = g_hash_table_lookup_int64(guild->channels, channel->parent_id);
+
+	gchar *guild_name = (guild) ? g_strdup_printf("%s: ", guild->name) : g_strdup("");
+	gchar *cat_name = (cat) ? g_strdup_printf("%s: ", cat->name) : g_strdup("");
+	gchar *channel_name = (channel) ? g_strdup(channel->name) : g_strdup("");
+
 	/* Don't re-add the channel to the same group */
 	if (discord_find_chat_in_group(da->account, id, group) == NULL) {
-		PurpleChat *chat = purple_chat_new(da->account, channel->name, components);
+		PurpleChat *chat = purple_chat_new(da->account, g_strdup_printf("%s%s%s", guild_name, cat_name, channel_name), components);
 		purple_blist_add_chat(chat, group, NULL);
 	} else {
 		g_hash_table_unref(components);
@@ -3554,7 +3562,7 @@ discord_add_group_dms_to_blist(DiscordAccount *da)
 		gchar *id_str = from_int(*id);
 
 		if (purple_blist_find_chat(da->account, id_str) == NULL) {
-			discord_add_channel_to_blist(da, channel, NULL);
+			discord_add_channel_to_blist(da, channel, NULL, NULL);
 		}
 
 		g_free(id_str);
@@ -3606,7 +3614,7 @@ discord_got_group_dm(DiscordAccount *da, JsonObject *data)
 		&& purple_account_get_bool(da->account, "populate-blist", TRUE)
 		&& purple_blist_find_chat(da->account, id) == NULL) {
 
-		discord_add_channel_to_blist(da, channel, NULL);
+		discord_add_channel_to_blist(da, channel, NULL, NULL);
 	}
 
 	g_free(id);
@@ -4073,7 +4081,7 @@ discord_process_dispatch(DiscordAccount *da, const gchar *type, JsonObject *data
 
 					/* must have READ_MESSAGES */
 					if ((permission & PERM_VIEW_CHANNEL)) {
-						discord_add_channel_to_blist(da, channel, NULL);
+						discord_add_channel_to_blist(da, channel, guild, NULL);
 					}
 				}
 			}
@@ -5205,6 +5213,8 @@ discord_grab_group(const char *guild_name, const char *category_name, const gcha
 		combined_name = g_strdup_printf("%s: %s", guild_name, category_name);
 	else
 		combined_name = g_strdup(guild_name);
+	g_free(combined_name);
+	combined_name = g_strdup(guild_name);
 
 	for (PurpleBlistNode *node = purple_blist_get_root(); node != NULL; node = purple_blist_node_get_sibling_next(node)) {
 		if (!PURPLE_BLIST_NODE_IS_GROUP(node)) {
@@ -5275,6 +5285,8 @@ discord_buddy_guild(DiscordAccount *da, DiscordGuild *guild)
 			g_free(category_id);
 			category_id = from_int(guild->id);
 		}
+		g_free(category_id);
+		category_id = from_int(guild->id);
 
 		gchar *namepref_id = g_strdup_printf("%" G_GUINT64_FORMAT "-abbr", guild->id);
 		const gchar *guild_name = purple_account_get_string(da->account, namepref_id, guild->name);
@@ -5286,7 +5298,7 @@ discord_buddy_guild(DiscordAccount *da, DiscordGuild *guild)
 		if (!group)
 			continue;
 
-		discord_add_channel_to_blist(da, channel, group);
+		discord_add_channel_to_blist(da, channel, guild, group);
 	}
 }
 
