diff --git a/src/player/mpvobject.cpp b/src/player/mpvobject.cpp
index 89e3650..beeb5af 100644
--- a/src/player/mpvobject.cpp
+++ b/src/player/mpvobject.cpp
@@ -30,7 +30,7 @@ MpvObject::MpvObject(QQuickItem * parent)
     //mpv::qt::set_option_variant(mpv, "input-cursor", "no");
 
     // Request hw decoding, just for testing.
-    mpv::qt::set_option_variant(mpv, "hwdec", "auto");
+    mpv::qt::set_option_variant(mpv, "hwdec", "off");
 
     //Cache
     //mpv::qt::set_option_variant(mpv, "cache", 8192);
