From fa3bf69571e45f9f279082994a986301be2b5c8e Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Tue, 24 Oct 2023 12:17:28 -0600
Subject: [PATCH 1/2] win32u: Bump AMD driver version.

---
 dlls/win32u/sysparams.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/win32u/sysparams.c b/dlls/win32u/sysparams.c
index e33ff6e60e5..8173031ae44 100644
--- a/dlls/win32u/sysparams.c
+++ b/dlls/win32u/sysparams.c
@@ -1229,7 +1229,7 @@ static BOOL write_gpu_to_registry( const struct gpu *gpu, const struct pci_id *p
         break;
     /* AMD */
     case 0x1002:
-        strcpy( buffer, "31.0.14051.5006" );
+        strcpy( buffer, "31.0.21902.5" );
         break;
     /* Nvidia */
     case 0x10de:
-- 
2.49.0

