From 1fe68bf424b52863ae454e28d65c86cb50e9e889 Mon Sep 17 00:00:00 2001
From: Arkadiusz Hiler <ahiler@codeweavers.com>
Date: Thu, 23 May 2024 16:58:09 +0300
Subject: [PATCH] win32u: Update Nvidia's DriverVersion to 999.99.

To avoid periodic bumps... at lest for some time.
---
 dlls/win32u/sysparams.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/win32u/sysparams.c b/dlls/win32u/sysparams.c
index 2b4a397af9b..e83e4b4f966 100644
--- a/dlls/win32u/sysparams.c
+++ b/dlls/win32u/sysparams.c
@@ -1406,7 +1406,7 @@ static void add_gpu( const struct gdi_gpu *gpu, void *param )
             break;
         /* Nvidia */
         case 0x10de:
-            strcpy( buffer, "31.0.15.3625" );
+            strcpy( buffer, "31.0.19.9999" );
             break;
         /* Default value for any other vendor. */
         default:
-- 
2.47.1

