diff --git a/Makefile.in b/Makefile.in
index b18c9077..01597b8f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -587,7 +587,7 @@ VKD3D_CONFIGURE_ARGS = \
 # LTO is enabled by default, but causes the build to fail. I havent't
 # investigated this yet.
 VKD3D_CFLAGS = -fno-lto
-VKD3D_LDFLAGS = -static-libgcc $(CROSSLDFLAGS)
+VKD3D_LDFLAGS = -static -static-libgcc -static-libstdc++ -Wl,-Bdynamic -lvkd3d-shader $(CROSSLDFLAGS)
 VKD3D_DEPENDS = vulkan-headers spirv-headers
 
 $(eval $(call rules-source,vkd3d,$(SRCDIR)/vkd3d))
