From 01e2fdaef957d82c57ec57e56c3b419954270834 Mon Sep 17 00:00:00 2001
From: Colin Kinloch <colin.kinloch@collabora.com>
Date: Sat, 29 Jun 2024 01:17:51 +0100
Subject: [PATCH] mangoapp: Use existing MANGOHUD_CONFIGFILE if set

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 59dec4f82..31bee291e 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -665,7 +665,7 @@ static void UpdateCompatEnvVars()
 	// A sane default here.
 	setenv( "GAMESCOPE_NV12_COLORSPACE", "k_EStreamColorspace_BT601", 0 );
 
-	if ( g_bLaunchMangoapp )
+	if ( g_bLaunchMangoapp && !getenv("MANGOHUD_CONFIGFILE") )
 	{
 		char szMangoConfigPath[ PATH_MAX ];
 		int nMangoConfigFd = gamescope::MakeTempFile( szMangoConfigPath, gamescope::k_szGamescopeTempMangoappTemplate );
